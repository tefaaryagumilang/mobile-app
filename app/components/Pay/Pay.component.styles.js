import * as styles from '../../styles/common.styles';
import {theme} from '../../styles/core.styles';
import {Dimensions} from 'react-native';
const {width} = Dimensions.get('window');

export default {
  container: [
    styles.contentContainerStyle,
    styles.containerWhiteStyle,
    {
      flex: 1,
      paddingVertical: 0
    }
  ],
  featureTitle: [
    styles.featureTitleStyle,
    styles.fontSizeLargeStyle
  ],
  featureSubTitle: styles.fontSizeNormalStyle,
  serviceContainer: {
    flex: 2,
    marginTop: 20,
    justifyContent: 'space-between'
  },
  serviceItemRow: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'center'
  },
  banner: {
    alignSelf: 'flex-end'
  },
  merchantPayContainer: {
    flexDirection: 'row',
    borderWidth: 1,
    borderBottomWidth: 2,
    borderColor: theme.grey,
    borderRadius: 5,
    marginBottom: 10,
    backgroundColor: theme.white,
    padding: 10,
    alignItems: 'center',
    marginVertical: 20
  },
  merchantPayRp: {
    borderRadius: 50,
    padding: 10,
    backgroundColor: theme.lightBlue,
    borderWidth: 1,
    borderColor: theme.white,
    alignItems: 'center',
  },
  rpText: {
    color: theme.white,
    fontWeight: 'bold',
  },
  merchantPayAt: {
    fontWeight: 'bold',
    fontSize: 15
  },
  merchantPayMargin: {
    margin: 10,
    width: 57 * width / 100,
  },
  merchantPayWith: {
    fontSize: 12,
    paddingTop: 5
  },
  merchantPayIcon: {
    alignItems: 'flex-end',
    paddingLeft: 10,
    width: 7 * width / 100,
  },
  arrowIcon: {
    color: theme.grey
  },
  iconRp: {
    color: theme.blueAmount,
    paddingHorizontal: 1
  }
};
