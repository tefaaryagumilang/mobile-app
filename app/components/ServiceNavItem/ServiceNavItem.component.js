import React from 'react';
import PropTypes from 'prop-types';
import {Text, View} from 'react-native';
import Touchable from '../../components/Touchable.component';
// import SimasIcon from '../../assets/fonts/SimasIcon';
import LayeredIcon from '../../components/LayeredIcon/LayeredIcon.component';
import styles from './ServiceNavItem.component.styles';
import noop from 'lodash/noop';

class ServiceNavItem extends React.Component {
  static propTypes = {
    onPress: PropTypes.func,
    highlightColor: PropTypes.string,
    highlightOpacity: PropTypes.number,
    text: PropTypes.string,
    iconName: PropTypes.string,
    disable: PropTypes.bool,
    iconSize: PropTypes.number,
    layers: PropTypes.array,
  };
  render () {
    const {onPress = noop, highlightColor = 'white', highlightOpacity = 0.5, text, iconName, disable = false, layers} = this.props;
    const enableStyles = disable ? styles.disable : styles.enable;
    return (
      <Touchable onPress={onPress} highlightColor={highlightColor} activeOpacity={highlightOpacity}>
        {iconName ?
          <View style={styles.itemContainer}>
            <View style={styles.icon}>
              <LayeredIcon layers={layers}/>
            </View>
            <View style={styles.textContainer}>
              <Text style={enableStyles}>{text}</Text>
            </View>
          </View>
          :
          <View style={styles.itemContainer}/>
        }
      </Touchable>
    );
  }
}

export default ServiceNavItem;
