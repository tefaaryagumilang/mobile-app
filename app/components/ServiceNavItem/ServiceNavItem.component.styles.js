import * as styles from '../../styles/common.styles';
import {viewportWidth} from '../../utils/device.util';
import {theme} from '../../styles/core.styles';

export default {
  itemContainer: {
    // justifyContent: 'center',
    alignItems: 'center',
    height: 70,
    width: (viewportWidth - styles.contentContainerStyle.padding * 2 - 2) / 4,
    // backgroundColor: 'yellow',
    marginBottom: 1,
    marginTop: 15,
  },
  itemContainerDisabled: {
    // justifyContent: 'center',
    alignItems: 'center',
    height: 70,
    width: (viewportWidth - styles.contentContainerStyle.padding * 2 - 2) / 4,
    marginBottom: 1,
    marginTop: 15,
    backgroundColor: theme.lightGrey
  },
  icon: {
    alignSelf: 'center',
    width: 40,
    height: 40,
  },
  textContainer: {
    alignSelf: 'center',
    width: (viewportWidth - styles.contentContainerStyle.padding * 2 - 2) / 4,
    height: 30,
    justifyContent: 'center'
  },
  enable: {
    color: theme.black,
    textAlign: 'center',
    fontSize: theme.fontSizeSmall
  },
  disable: {
    color: theme.grey,
    textAlign: 'center',
    fontSize: theme.fontSizeSmall
  }
};
