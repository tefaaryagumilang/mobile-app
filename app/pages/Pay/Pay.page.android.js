import React, {Component} from 'react';
import PropTypes from 'prop-types';
import PayComponent from '../../components/Pay/Pay.component';
import {setupPayment, setupPaymentGopay, getBillpayHistory, payByHistory, deleteBillpayHistory, tokenInvoiceHistory} from '../../state/thunks/common.thunks';
import {connect} from 'react-redux';
import result from 'lodash/result';
import {goToSDK as goToSDKThunk, checkEULA} from '../../state/thunks/qrpayment.thunk';
import {PermissionsAndroid} from 'react-native';
import {Toast} from '../../utils/RNHelpers.util';
import {language} from '../../config/language';
import {genericBillerNavigate} from '../../utils/genericBiller.util';
import {getFilteredBillerDataRevamp, upperCase} from '../../utils/transformer.util';
import {getFavBiller} from '../../state/thunks/dashboard.thunks';
import isEmpty from 'lodash/isEmpty';
import {messageBillerNK} from '../../state/thunks/genericBill.thunks';


const mapStateToProps = (state) => ({
  currentLanguage: state.currentLanguage,
  userApiKey: result(state, 'userApiKey', ''),
  billerMenuOrderRevamp: result(state, 'config.billerMenuOrderRevamp', []),
  billpayHistory: result(state, 'billpayHistory', {}),
  billerConfig: result(state, 'billerConfig', {}),
  haveEmoney: result(state, 'generateCode', {}),
  flagLKDPurchase: upperCase(result(state, 'config.flag.flagLKDPurchase', 'INACTIVE')) === upperCase('ACTIVE'),
  transactionTypeLKD: result(state, 'config.transactionTypeLKD', {}),
  drawer: result(state, 'drawer', false),
  isLogin: !isEmpty(result(state, 'user', {})),
  cifString: result(state, 'user.profile.customer.cifCode', ''),
});

const mapDispatchToProps = (dispatch) => ({
  setupPaymentGopayForNavigation: (route, isGopay) => {
    dispatch(getFavBiller());
    dispatch(setupPaymentGopay(route, isGopay));
  },
  setupPaymentForNavigation: (route, billerTypeId) => {
    dispatch(getFavBiller());
    dispatch(setupPayment(route, billerTypeId));
  },
  goToQrPayment: () => {
    dispatch(goToSDKThunk());
  },
  checkEULAandNavigate: () => {
    dispatch(checkEULA());
  },
  getBillpayHistory: () => dispatch(getBillpayHistory()),
  deleteBillpayHistory: (history) => () => dispatch(deleteBillpayHistory(history)),
  goToIndex: (biller, subscriberNo, description) => () => {
    dispatch(getFavBiller());
    dispatch(payByHistory(biller, subscriberNo, description));
  },
  goToBiller: (biller) => {
    dispatch(getFavBiller());
    genericBillerNavigate(dispatch, biller);
  },
  toPushInvoiceHistory: () => {
    dispatch(tokenInvoiceHistory());
  },
  messageNK: () => {
    dispatch(messageBillerNK());
  }
});

class PayPage extends Component {

  static propTypes = {
    navigation: PropTypes.object,
    setupPaymentForNavigation: PropTypes.func,
    setupPaymentGopayForNavigation: PropTypes.func,
    goToQrPayment: PropTypes.func,
    userApiKey: PropTypes.string,
    checkEULAandNavigate: PropTypes.func,
    billerMenuOrderRevamp: PropTypes.array,
    goPayNavigateTo: PropTypes.func,
    getBillpayHistory: PropTypes.func,
    billpayHistory: PropTypes.object,
    billerConfig: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    goToIndex: PropTypes.func,
    deleteBillpayHistory: PropTypes.func,
    goToBiller: PropTypes.func,
    haveEmoney: PropTypes.oneOfType([
      PropTypes.object,
      PropTypes.bool
    ]),
    flagLKDPurchase: PropTypes.bool,
    transactionTypeLKD: PropTypes.object,
    toPushInvoiceHistory: PropTypes.func,
    getFavBiller: PropTypes.func,
    drawer: PropTypes.bool,
    isLogin: PropTypes.bool,
    cifString: PropTypes.string,
    messageNK: PropTypes.func
  }

  onBuyMobileTopTop = () => {
    this.props.setupPaymentForNavigation('MobileTopup');
  }

  navigateTo = (routeName, billerTypeId) => () => {
    this.props.setupPaymentForNavigation(routeName, billerTypeId);
  }

  goPayNavigateTo = (routeName, isGopay) => () => {
    this.props.setupPaymentGopayForNavigation(routeName, isGopay);
  }

  goToBiller = (billerName) => () => {
    const {billerConfig} = this.props;
    const biller = getFilteredBillerDataRevamp(billerConfig, billerName)[0];
    this.props.goToBiller(biller);
  }

  onQR = () => {
    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.CAMERA).
      then((status) => {
        if ([PermissionsAndroid.RESULTS.GRANTED, true].includes(status)) {
          this.props.checkEULAandNavigate();
        } else {
          Toast.show(language.PERMISSION_ERROR__CONTACTS);
        }
      }).catch(() => {
        Toast.show(language.PERMISSION_ERROR__CONTACTS);
      });
  }
  messageNK = () => () => {
    Toast.show(language.EMONEY__CANNOTGETBILL, Toast.SHORT);
  }

  componentWillMount () {
    this.props.getBillpayHistory();
  }

  render () {
    const {goToQrPayment, billerMenuOrderRevamp, billpayHistory, billerConfig, goToIndex, cifString, isLogin,
      deleteBillpayHistory, getBillpayHistory, haveEmoney, flagLKDPurchase, transactionTypeLKD, drawer, toPushInvoiceHistory} = this.props;
    const billerList = result(billerConfig, 'billerList', []);
    return (
      <PayComponent
        serviceList={this.serviceList}
        navigateTo={this.navigateTo}
        onBuyMobileTopTop={this.onBuyMobileTopTop}
        goToQrPayment={goToQrPayment}
        onQR={this.onQR}
        goPayNavigateTo={this.goPayNavigateTo}
        billerMenuOrder={billerMenuOrderRevamp}
        billpayHistory={billpayHistory}
        billerConfig={billerConfig}
        billerList={billerList}
        goToIndex={goToIndex}
        deleteBillpayHistory={deleteBillpayHistory}
        goToBiller={this.goToBiller}
        reloadHistory={getBillpayHistory}
        haveEmoney={haveEmoney}
        flagLKDPurchase= {flagLKDPurchase}
        transactionTypeLKD= {transactionTypeLKD}
        toPushInvoiceHistory={toPushInvoiceHistory}
        drawer = {drawer}
        isLogin={isLogin}
        cifString={cifString}
        messageNK={this.messageNK}
      />
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PayPage);
