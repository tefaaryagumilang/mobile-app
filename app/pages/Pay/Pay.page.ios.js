import React, {Component} from 'react';
import PropTypes from 'prop-types';
import PayComponent from '../../components/Pay/Pay.component';
import {setupPayment, setupPaymentGopay,
  getBillpayHistory, payByHistory, deleteBillpayHistory, tokenInvoiceHistory} from '../../state/thunks/common.thunks';
import {connect} from 'react-redux';
import result from 'lodash/result';
import {goToSDK as goToSDKThunk, checkEULA} from '../../state/thunks/qrpayment.thunk';
import Permissions from 'react-native-permissions';
import {Toast} from '../../utils/RNHelpers.util';
import {language} from '../../config/language';
import {genericBillerNavigate} from '../../utils/genericBiller.util';
import {getFilteredBillerDataRevamp, upperCase} from '../../utils/transformer.util';
import {getFavBiller} from '../../state/thunks/dashboard.thunks';
import isEmpty from 'lodash/isEmpty';
import {messageBillerNK} from '../../state/thunks/genericBill.thunks';
import {Platform} from 'react-native';


let PermissionsAndroid;

if (Platform.OS === 'android') {
  PermissionsAndroid = require('react-native').PermissionsAndroid;
}

const mapStateToProps = (state) => ({
  currentLanguage: state.currentLanguage,
  userApiKey: result(state, 'userApiKey', ''),
  billerMenuOrderRevamp: result(state, 'config.billerMenuOrderRevamp', []),
  billpayHistory: result(state, 'billpayHistory', {}),
  billerConfig: result(state, 'billerConfig', {}),
  haveEmoney: result(state, 'generateCode', {}),
  flagLKDPurchase: upperCase(result(state, 'config.flag.flagLKDPurchase', 'INACTIVE')) === upperCase('ACTIVE'),
  transactionTypeLKD: result(state, 'config.transactionTypeLKD', {}),
  drawer: result(state, 'drawer', false),
  isLogin: !isEmpty(result(state, 'user', {})),
  cifString: result(state, 'user.profile.customer.cifCode', ''),
});

const mapDispatchToProps = (dispatch) => ({
  setupPaymentGopayForNavigation: (route, isGopay) => {
    dispatch(getFavBiller());
    dispatch(setupPaymentGopay(route, isGopay));
  },
  setupPaymentForNavigation: (route, billerTypeId) => {
    dispatch(getFavBiller());
    dispatch(setupPayment(route, billerTypeId));
  },
  goToQrPayment: () => {
    dispatch(goToSDKThunk());
  },
  checkEULAandNavigate: () => {
    dispatch(checkEULA());
  },
  getBillpayHistory: () => dispatch(getBillpayHistory()),
  deleteBillpayHistory: (history) => () => dispatch(deleteBillpayHistory(history)),
  goToIndex: (biller, subscriberNo, description) => () => {
    dispatch(payByHistory(biller, subscriberNo, description));
    dispatch(getFavBiller());
  },
  goToBiller: (biller) => {
    dispatch(getFavBiller());
    genericBillerNavigate(dispatch, biller);
  },
  toPushInvoiceHistory: () => {
    dispatch(tokenInvoiceHistory());
  },
  messageNK: () => {
    dispatch(messageBillerNK());
  }
});

class PayPage extends Component {

  static propTypes = {
    navigation: PropTypes.object,
    setupPaymentForNavigation: PropTypes.func,
    setupPaymentGopayForNavigation: PropTypes.func,
    goToQrPayment: PropTypes.func,
    userApiKey: PropTypes.string,
    checkEULAandNavigate: PropTypes.func,
    billerMenuOrderRevamp: PropTypes.array,
    goPayNavigateTo: PropTypes.func,
    getBillpayHistory: PropTypes.func,
    billpayHistory: PropTypes.object,
    billerConfig: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    goToIndex: PropTypes.func,
    deleteBillpayHistory: PropTypes.func,
    goToBiller: PropTypes.func,
    haveEmoney: PropTypes.oneOfType([
      PropTypes.bool,
      PropTypes.object
    ]),
    flagLKDPurchase: PropTypes.bool,
    transactionTypeLKD: PropTypes.object,
    toPushInvoiceHistory: PropTypes.func,
    getFavBiller: PropTypes.func,
    drawer: PropTypes.bool,
    isLogin: PropTypes.bool,
    cifString: PropTypes.string,
    messageNK: PropTypes.func
  }

  onBuyMobileTopTop = () => {
    this.props.setupPaymentForNavigation('MobileTopup');
  }

  goToBiller = (billerName) => () => {
    const {billerConfig} = this.props;
    const biller = getFilteredBillerDataRevamp(billerConfig, billerName)[0];
    this.props.goToBiller(biller);
  }

  navigateTo = (routeName, billerTypeId) => () => {
    this.props.setupPaymentForNavigation(routeName, billerTypeId);
  }

  goPayNavigateTo = (routeName, isGopay) => () => {
    this.props.setupPaymentGopayForNavigation(routeName, isGopay);
  }

  onQR = () => {
    if (Platform.OS === 'android') {
      PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.CAMERA).then((response) => {
        if (response === 'granted') {
          this.props.checkEULAandNavigate();
        } else {
          PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.CAMERA).then((response) => {
            if (response === 'granted') {
              this.props.checkEULAandNavigate();
            } else {
              Toast.show(language.PERMISSION__CAMERA, Toast.LONG);
            }
          });
        }
      });
    } else { 
      Permissions.check('ios.permission.CAMERA').then((response) => {
        if (response === 'granted') {
          this.props.checkEULAandNavigate();
        } else {
          Permissions.request('ios.permission.CAMERA').then((response) => {
            if (response === 'granted') {
              this.props.checkEULAandNavigate();
            } else {
              Toast.show(language.PERMISSION__CAMERA, Toast.LONG);
            }
          });
        }
      });
    }
  }
  messageNK = () => () => {
    Toast.show(language.EMONEY__CANNOTGETBILL, Toast.SHORT);
  }
  componentWillMount () {
    this.props.getBillpayHistory();
  }

  render () {
    const {goToQrPayment, billerMenuOrderRevamp, billpayHistory, billerConfig, goToIndex,
      deleteBillpayHistory, getBillpayHistory, cifString, isLogin, haveEmoney, flagLKDPurchase, transactionTypeLKD, toPushInvoiceHistory, drawer} = this.props;
    const billerList = result(billerConfig, 'billerList', []);
    return (
      <PayComponent
        serviceList={this.serviceList}
        navigateTo={this.navigateTo}
        onBuyMobileTopTop={this.onBuyMobileTopTop}
        goToQrPayment={goToQrPayment}
        onQR={this.onQR}
        goPayNavigateTo={this.goPayNavigateTo}
        billerMenuOrder={billerMenuOrderRevamp}
        billpayHistory={billpayHistory}
        billerConfig={billerConfig}
        billerList={billerList}
        goToIndex={goToIndex}
        deleteBillpayHistory={deleteBillpayHistory}
        goToBiller={this.goToBiller}
        reloadHistory={getBillpayHistory}
        haveEmoney={haveEmoney}
        flagLKDPurchase= {flagLKDPurchase}
        transactionTypeLKD= {transactionTypeLKD}
        toPushInvoiceHistory={toPushInvoiceHistory}
        drawer={drawer}
        isLogin={isLogin}
        cifString={cifString}
        messageNK={this.messageNK}
      />
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PayPage);
