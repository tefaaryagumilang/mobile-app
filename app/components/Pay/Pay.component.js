import React from 'react';
import PropTypes from 'prop-types';
import {ScrollView, Text, View} from 'react-native';
import {language} from '../../config/language';
import ServiceNavItem from '../ServiceNavItem/ServiceNavItem.component';
import PaymentHistory from '../PaymentHistory/PaymentHistory.component';
import styles from './Pay.component.styles';
import noop from 'lodash/noop';
import {generateBillerMenu, generateBillerMenuBeforeLogin} from '../../utils/transformer.util';
import startsWith from 'lodash/startsWith';

class Pay extends React.Component {

  static propTypes = {
    serviceList: PropTypes.array,
    onBuyMobileTopTop: PropTypes.func,
    goToQrPayment: PropTypes.func,
    navigateTo: PropTypes.func,
    goPayNavigateTo: PropTypes.func,
    onQR: PropTypes.func,
    billerMenuOrder: PropTypes.array,
    billpayHistory: PropTypes.object,
    billerList: PropTypes.array,
    goToIndex: PropTypes.func,
    deleteBillpayHistory: PropTypes.func,
    goToBiller: PropTypes.func,
    reloadHistory: PropTypes.func,
    toGenerateMain: PropTypes.func,
    flagLKDPurchase: PropTypes.bool,
    transactionTypeLKD: PropTypes.object,
    toPushInvoiceHistory: PropTypes.func,
    drawer: PropTypes.bool,
    isLogin: PropTypes.bool,
    cifString: PropTypes.string,
    messageNK: PropTypes.func
  }

  render () {
    const {onBuyMobileTopTop, navigateTo = noop, goPayNavigateTo = noop, onQR = noop, billerMenuOrder,
      billpayHistory, billerList, goToIndex, deleteBillpayHistory, goToBiller, reloadHistory, drawer, cifString, messageNK,  isLogin} = this.props;
    const isVerified = !startsWith(cifString, 'NK');
    const serviceList = isLogin ? generateBillerMenu(billerMenuOrder, {navigateTo, onBuyMobileTopTop, onQR, goPayNavigateTo, goToBiller, messageNK}, isVerified)
      : generateBillerMenuBeforeLogin(billerMenuOrder, {navigateTo, onBuyMobileTopTop, onQR, goPayNavigateTo, goToBiller});
    serviceList.push({
      iconName: '',
      title: '',
    });
    const serviceList2 = serviceList.slice(0, serviceList.length - 1);
    const serviceNavItems = serviceList2.map((item, index) => {
      const {iconName, title, onPress, disable, iconSize, layers} = item;
      return (<ServiceNavItem key={`service_${index}`} iconName={iconName} iconSize={iconSize}
        text={title} onPress={onPress} disable={disable} layers={layers}/>);
    });
    return (
      <ScrollView style={styles.container}>
        <View style={styles.serviceContainer}>
          <Text style={styles.featureTitle}>{language.PAY_BILLS__TITLE}</Text>
          <View style={styles.serviceItemRow}>
            {serviceNavItems}
          </View>
        </View>
        <PaymentHistory billpayHistory={billpayHistory} billerList={billerList}
          goToIndex={goToIndex} deleteBillpayHistory={deleteBillpayHistory}
          drawer={drawer} reloadHistory={reloadHistory} />
      </ScrollView>
    );
  }
}

export default Pay;
