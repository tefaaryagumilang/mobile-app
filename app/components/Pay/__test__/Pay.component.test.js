import React from 'react';
import renderer from 'react-test-renderer';
import Pay from '../Pay.component';
import ServiceNavItem from '../../ServiceNavItem/ServiceNavItem.component';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

describe('OnboardingJourney Component: AccountSettingsView', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Pay/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('render 2 ServiceNavItem components', () => {
    const navigateTo = jest.fn();
    const billerMenuOrder = [0];
    const wrapper = shallow(<Pay navigateTo={navigateTo} billerMenuOrder={billerMenuOrder}/>);
    const items = wrapper.find(ServiceNavItem);
    items.first().simulate('press');
    expect(navigateTo).toBeCalled();
  });
});
