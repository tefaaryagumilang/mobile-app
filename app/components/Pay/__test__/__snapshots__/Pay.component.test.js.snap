// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingJourney Component: AccountSettingsView renders correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "padding": 20,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
      Object {
        "flex": 1,
        "paddingVertical": 0,
      },
    ]
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 2,
          "justifyContent": "space-between",
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 12,
              "fontWeight": "700",
            },
            Object {
              "color": "#000",
              "fontSize": 18,
            },
          ]
        }
      >
        Bayar / Isi Ulang
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 30,
            "marginTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#949494",
              "flex": 3,
              "height": 1,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 2,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#949494",
                "fontFamily": "roboto",
                "fontSize": 12,
                "textAlign": "center",
              }
            }
          >
            ATAU
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#949494",
              "flex": 3,
              "height": 1,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 18,
            },
            Object {
              "color": "#000000",
              "fontFamily": "roboto",
            },
          ]
        }
      >
        Cari dari transaksi sebelumnya
      </Text>
      <View
        style={
          Object {
            "borderColor": "#949494",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
            "marginTop": 15,
            "paddingHorizontal": 10,
            "paddingVertical": 15,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          maxLength={20}
          onChangeText={[Function]}
          placeholder="Nama biller / Nomor pelanggan"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#000000",
              "flex": 1,
              "fontSize": 16,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 20,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "roboto",
                },
                Object {
                  "fontFamily": "simas",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î§´
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 30,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
